include/wil/registry_helpers.h

```diff
@@ -0,0 +1,1030 @@
+                {
+                    return REG_SZ;
+                }
+#endif // #if defined(__WIL_OLEAUTO_H_)
+
+#if defined(__WIL_OLEAUTO_H_STL)
+                constexpr DWORD set_value_type(const ::wil::shared_bstr&) WI_NOEXCEPT
+                {
+                    return REG_SZ;
+                }
+#endif // #if defined(__WIL_OLEAUTO_H_STL)
+
+#if defined(__WIL_OBJBASE_H_)
+                constexpr DWORD set_value_type(const ::wil::unique_cotaskmem_string&) WI_NOEXCEPT
+                {
+                    return REG_SZ;
+                }
+#endif // defined(__WIL_OBJBASE_H_)
+
+#if defined(__WIL_OBJBASE_H_STL)
+                constexpr DWORD set_value_type(const ::wil::shared_cotaskmem_string&) WI_NOEXCEPT
+                {
+                    return REG_SZ;
+                }
+#endif // #if defined(__WIL_OBJBASE_H_STL)
+            }
+
+            template <typename err_policy = ::wil::err_exception_policy>
+            class reg_view_t
+            {
+            public:
+                explicit reg_view_t(HKEY key) WI_NOEXCEPT :
+                    m_key(key)
+                {
+                }
+                ~reg_view_t() = default;
+                reg_view_t(const reg_view_t&) = delete;
+                reg_view_t& operator=(const reg_view_t&) = delete;
+                reg_view_t(reg_view_t&&) = delete;
+                reg_view_t& operator=(reg_view_t&&) = delete;
+
+                typename err_policy::result open_key(_In_opt_ PCWSTR subKey, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
+                {
+                    constexpr DWORD zero_ulOptions{ 0 };
+                    const auto error = ::RegOpenKeyExW(m_key, subKey, zero_ulOptions, get_access_flags(access), hkey);
+                    return err_policy::HResult(HRESULT_FROM_WIN32(error));
+                }
+
+                ::wil::unique_hkey open_unique_key(_In_opt_ PCWSTR subKey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
+                {
+                    ::wil::unique_hkey local_key{};
+                    open_key(subKey, &local_key, access);
+                    return local_key;
+                }
+
+#if defined(__WIL_WINREG_STL)
+                ::wil::shared_hkey open_shared_key(_In_opt_ PCWSTR subKey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
+                {
+                    ::wil::shared_hkey local_key{};
+                    open_key(subKey, &local_key, access);
+                    return local_key;
+                }
+#endif // #if defined(__WIL_WINREG_STL)
+
+                typename err_policy::result create_key(PCWSTR subKey, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
+                {
+                    *hkey = nullptr;
+
+                    constexpr DWORD zero_Reserved{ 0 };
+                    constexpr LPWSTR null_lpClass{ nullptr };

```

Comment:
L no longer needed?

and if we're following os guidelines, avoid hungarian?

Reply:
I was pasting from the SDK header. but I should really clean these up.

Reply:
Thanks. cleaned this up

Reply:
Seems resolved; feel free to reopen if I'm wrong :)